### PR TITLE
Simplify logic by unnesting if expressions.

### PR DESCRIPTION
**Pull request by GunnarFarneback**
This PR simplifies the implementation of `cron_or_api_build(api, pr, registry)` by repeatedly applying the rewrite from
```
if x
    y
else
    z
end
return nothing
```
to
```
if !x
    z
    return nothing
end
y
return nothing
```
which dramatically reduces the nesting depth of the code and makes the logic easier to follow. There should be no functional differences.

Review tip: The only effective way to review is probably to apply these transformations yourself and see if you end up with the same code.

### CODE REVIEW

File: src/AutoMerge/cron.jl [hunk 1]
```diff
@@ -271,152 +271,158 @@ function cron_or_api_build(api::GitHub.GitHubAPI,
...
-                                                                   all_statuses;
-                                                                   auth = auth)
-                    _checkruns_good = all_specified_check_runs_passed(api, registry,
-                                                                      pr,
-                                                                      passed_pr_head_sha,
-                                                                      all_check_runs;
-                                                                      auth = auth)
-                    if _statuses_good && _checkruns_good
-                        if pr_has_no_blocking_comments(api, registry, pr; auth = auth)
-                            "Pull request: $(pr_number). "
```
Comment (GunnarFarneback) [inline src/AutoMerge/cron.jl:319]: Why are there loose strings here? Was this supposed to be an `@info`?

Comment (DilumAluthge) [reply src/AutoMerge/cron.jl:319]: Yeah that should be an `@info`.

Comment (GunnarFarneback) [reply src/AutoMerge/cron.jl:319]: Actually there's an `@info` with exactly those strings a few lines below, so these copies should probably just be deleted.

**Comment by ericphanson** [comment]
Thanks @GunnarFarneback! I’ll review both of these in the next few days.

**Review by ericphanson — APPROVED** [review]
looks good! Much easier to follow, and it makes sense as sequence of checks to see if we should reject (and if so, to reject immediately).

**Comment by ericphanson** [comment]
Let's check CI:

bors try

**Comment by ericphanson** [comment]
bors merge

**Pull request merged**